File: benchmarks/fixtures/webapp_csharp/Validators/UserValidator.cs
namespace Webapp.Validators;

using Webapp.Errors;
using Webapp.Util;

/// <summary>
/// Validates user-facing input.
/// </summary>
public class UserValidator
{
    private static readonly Logger Log = Logger.GetLogger("validators.user");

    public void Validate(string email)
    {
        Log.Info("Validating user input");
        if (string.IsNullOrEmpty(email))
        {
            throw new ValidationException("email is required");
        }
    }

    public void ValidateLogin(string email, string password)
    {
        Validate(email);
        if (password == null || password.Length < 6)
        {
            throw new ValidationException("password too short");
        }
    }
}

File: benchmarks/fixtures/webapp_cpp/validators/user_validator.cpp
#include "validators/user_validator.h"

#include <string>

#include "errors/exceptions.h"
#include "util/logger.h"

namespace webapp {
namespace validators {

UserValidator::UserValidator() : log_(util::Logger::get_logger("validators.user")) {}

/// Reject an empty email address.
void UserValidator::validate(const std::string& email) const {
    log_.info("Validating user input");
    if (email.empty()) {
        throw errors::ValidationError("email is required");
    }
}

/// Reject an empty email or a too-short password.
void UserValidator::validate_login(const std::string& email, const std::string& password) const {
    validate(email);
    if (password.size() < 6) {
        throw errors::ValidationError("password too short");
    }
}

}  // namespace validators
}  // namespace webapp
